### ISSUE TITLE
Discuss: Release object store `0.12.0` (breaking) or `0.11.2` (non breaking API) around Dec 15, 2024

### ISSUE THREAD

**Issue by alamb** [apache/arrow-rs#6596]
**Is your feature request related to a problem or challenge? Please describe what you are trying to do.**
Follow on to https://github.com/apache/arrow-rs/issues/6121

object_store `0.11.1` was released Oct 20, 2024

Per release schedule here https://github.com/apache/arrow-rs/tree/master?tab=readme-ov-file#object_store-crate we aim for 
2 months release

**Describe the solution you'd like**
- [ ] Update version + make release noted 
- [ ] Prepare release candidate + begin vote thread
- [] Release to crates.io

**Describe alternatives you've considered**
1. Another incremental release
2. A major release including breaking API changes (e.g. to include https://github.com/apache/arrow-rs/pull/6266)

**Additional context**
I don't have any preference, though I personally think https://github.com/apache/arrow-rs/pull/6266 is a nice to have

**Comment by tustvold** [comment]
I agree that at the moment we probably don't have enough to justify a breaking release.

Some other things that might be breaking, but not sure if/when we will see anything in that space

* https://github.com/apache/arrow-rs/issues/5351
* https://github.com/apache/arrow-rs/issues/6587
* https://github.com/apache/arrow-rs/issues/6055

**Comment by alamb** [comment]
I propose that we plan to make the next object_store release 0.11.2 (non breaking) and try to make it next week. Please let me know if you have ojections

**Comment by tustvold** [comment]
I'd like to propose we proceed as planned with a non-breaking release next week, and then do a breaking release late February in advance of the scheduled March breaking arrow release.

This would mean we can start incorporating breaking changes into main in a week's time.

If there are major non-breaking features that land during that time, we can consider cutting another 0.11 patch release, presuming the contributor is willing to backport their changes.

**Comment by alamb** [comment]
I think @tustvold 's plan sounds good. Unless there objections I'll update the tickets to reflect that

**Comment by benesch** [comment]
Sounds great to me as well! Eager to be able to incorporate this into SlateDB.

**Comment by alamb** [comment]
I filed a PR to add documentation and two tickets to track the next two releases
- https://github.com/apache/arrow-rs/pull/6904
- https://github.com/apache/arrow-rs/issues/6902
- https://github.com/apache/arrow-rs/issues/6903

**Comment by criccomini** [comment]
🔥 _very_ excited to incorporate the new CAS feature into SlateDB. 😁
